File: 2344-minimum-deletions-to-make-array-divisible/2344-minimum-deletions-to-make-array-divisible.py
class Solution:
    def minOperations(self, nums: List[int], numsDivide: List[int]) -> int:
        
        div = reduce(math.gcd , numsDivide)
        nums.sort()
        for i, n in enumerate(nums):
            if div % n==0: 
                return i 
        return -1 

File: 2344-minimum-deletions-to-make-array-divisible/2344-minimum-deletions-to-make-array-divisible.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, vector<int>& numsDivide) {
     

        int a = numsDivide[0] ; 
        for(int i =1; i < numsDivide.size() ; i++) 
            a = gcd(a , numsDivide[i]) ; 

        sort(nums.begin() , nums.end()); 
        for(int j=0 ; j< nums.size() ; j++)
            if( a % nums[j] ==0) return j ; 
        return -1; 


        
        
    }
};